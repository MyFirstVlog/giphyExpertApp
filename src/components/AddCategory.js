import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategorias}) => {

    const [inputValue, setInputValue] = useState('')

    const handleChange = (e)=>{
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()

        if(inputValue.trim().length >= 2){
            setCategorias((actuales) => [inputValue,...actuales] )
            setInputValue('')
        }
    }

    return (
        <form onSubmit= {handleSubmit}>
            <input
                placeholder = 'Que desea buscar?'
                type="text"
                value={inputValue}
                onChange={handleChange}
            >
            </input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategorias : PropTypes.func.isRequired
}