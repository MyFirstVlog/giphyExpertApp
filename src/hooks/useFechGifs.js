import { useState, useEffect } from 'react';
import { getGifs } from "../helpers/GetGifs"



export const useFecthGifs = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {

        getGifs(category)
            .then(gifs => {

                setstate({
                    data: gifs,
                    loading: false
                })



            })


    }, [category])

    return state
}