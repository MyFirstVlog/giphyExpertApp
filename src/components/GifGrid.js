import React from 'react'
import PropTypes from 'prop-types'
import { getGifs } from '../helpers/GetGifs'
import { useFecthGifs } from '../hooks/useFechGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({ category }) => {

    const {data:images,loading} = useFecthGifs(category)

    /**
     * recordar que gifs => setimages(gifs) -> setimages
     * 
     * <GifGridItem key={img.id} {...img} --> manda la info de img como props independientes
     * 
     * const {data:images,loading} -> para renombrar variables
     */



    return (
        <>
            <h3>{category}</h3>

            {loading && <p className='animate__animated animate__jackInTheBox'>Loading...</p>}

            <div className='card-grid'>
                 {images.map(img => {
                     return <GifGridItem key={img.id} {...img} />
                })}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
} 