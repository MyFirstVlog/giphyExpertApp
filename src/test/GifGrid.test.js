import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../components/GifGrid';
import { useFecthGifs } from '../hooks/useFechGifs';
jest.mock('../hooks/useFechGifs') //finge llamada a ese archivo


describe('Pruebas con el componente <GetGrid />', () => {
    const category = 'CR7'

    // beforeEach(()=> {
    //     jest.clearAllMocks()
    //     wrapper = shallow(<GifGrid category={category} />)
    // })

    test('hacer el match del componente ', () => {

        useFecthGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper).toMatchSnapshot()
    })

    test('cuando useFetch() regresa info, verificar los items ', () => {
        //debemos hacer mock, como si mi custom hook regresara la info cuando ya la cargo 
        const gifs = [{
            id:'3425345',
            url:'http://localhost.com/images',
            title:'El Bicho'
        }]

        useFecthGifs.mockReturnValue({
            data:gifs,
            loading: false
        })
        
        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    
    
    })

})
