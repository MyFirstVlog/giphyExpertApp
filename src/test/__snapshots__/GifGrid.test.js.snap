// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GetGrid /> cuando useFetch() regresa info, verificar los items  1`] = `
<Fragment>
  <h3>
    CR7
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="3425345"
      key="3425345"
      title="El Bicho"
      url="http://localhost.com/images"
    />
  </div>
</Fragment>
`;

exports[`Pruebas con el componente <GetGrid /> hacer el match del componente  1`] = `
<Fragment>
  <h3>
    CR7
  </h3>
  <p
    className="animate__animated animate__jackInTheBox"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
